client/src/components/InputField.tsx

```diff
@@ -4,9 +4,16 @@ import { SendHorizontal, Loader2 } from "lucide-react";
 type InputFieldProps = {
   onSend: (message: string) => void;
   disabled?: boolean;
+  gameOver?: boolean;
+  onRestart?: () => void;
```

Comment:
Is that a type decoration? Never seen that before! I'm guessing it means `this is a function that takes no arguments and returns void`. Pretty cool.

Reply:
yep!